colossalai/inference/core/request_handler.py

```diff
@@ -1,71 +1,208 @@
 from typing import List
 
+import torch
+from transformers.configuration_utils import PretrainedConfig
+
+from colossalai.inference.config import InferenceConfig
+from colossalai.inference.kv_cache import KVCacheManager
+from colossalai.inference.logit_processors import logit_processor
+from colossalai.inference.sampler import *
 from colossalai.inference.struct import BatchInfo, Sequence
 
 
+class RunningList:
+    """
+    RunningList is an structure for recording the running sequences, contains prefill and decoding list.
+    Prefilling samples will be hold until the actual ratio of prefill samples versus decoding samples exceeds ratio.
+
+    Args:
+        prefill_ratio: (float) A ratio for determing whether to perform prefill or not.
+        prefill: (List) List that contains default inputs, defaults to [].
+    """
+
+    def __init__(self, prefill_ratio: str, prefill: List[Sequence] = []):
+        self.prefill_ratio = prefill_ratio
+        self.decoding: List[Sequence] = []
+        self.prefill: List[Sequence] = prefill
+
+    def append(self, seq: Sequence):
+        # add seq to prefilling list first.
+        self.prefill.append(seq)
+
+    def find_seq(self, request_id):
+        for seq in self.decoding:
+            if request_id == seq.request_id:
+                return seq
+        for seq in self.prefill:
+            if request_id == seq.request_id:
+                return seq
+        return None
+
+    def remove(self, seq):
+        try:
+            self.decoding.remove(seq)
+        except:
+            self.prefill.remove(seq)
```

Comment:
this looks a bit weird. shouldn't it be like:

```python
if seq in self.decoding:
    self.decoding.remove(seq)
elif seq in self.prefill:
   self.prefill.remove(seq)
else:
   raise ValueError(...)
```

Reply:
fixed